src/misc/index.ts

```diff
@@ -0,0 +1 @@
+export const portalSymbol = Symbol();
```

Comment:
not a big fan of this file.  I would define it in `Component.ts`, since you use it in that file anyway.

Also, it should be something like this:
```js
Symbol('portal');
```
for a better description of its use